packages/controller/lib/setup/setupUpload.js

```diff
@@ -601,7 +601,8 @@ function Upload(options) {
             return target;
         }
         for (const attr of Object.keys(additional)) {
-            if (attr === 'title' || attr === 'schedule' || attr === 'restartSchedule' || attr === 'mode' || attr === 'loglevel' || attr === 'enabled' || attr === 'custom') {
+            // preserve these attributes, except, they werde undefined before
+            if (attr === 'title' || attr === 'titleLang' || attr === 'schedule' || attr === 'restartSchedule' || attr === 'mode' || attr === 'loglevel' || attr === 'enabled' || attr === 'custom') {
```

Comment:
If you want to "refactor" I think it could be time to create a "preservedFieldNames =[...]" and "if preservedFieldNames.includes(attr)" instead of that growing if list :)

Reply:
On the other hand the question is if admin should not better "write" title if changed manually and use  that to display instead oiverwriting titleLang ... @GermanBluefox ?!

Reply:
Lets clarify the admin question first, if we will do it in controller I will refactor it a bit

Reply:
@GermanBluefox ? The question is how Admin should change the "title" if someone renames an adapter instance via admin. right now "titleLang" is overwritten and the next "iob upload adaptername" will restore the io-package version again.

So should we prevent this restore here in conteroller or should not admin better write the changed title in an other property which is displayed with precedence over titleLang?

Reply:
IMHO, admin should write titleLang. And title must disappier at all.

Reply:
So `iob upload` should override a user-defined title?

Reply:
Is admin setting then titleLang as an "Object" ith only "whatever language" or as a string? If it overrides as aa string: "upload command overrides only in titleLang is not a string"?

Reply:
Would be good to know, then we could implement some custom logic

Reply:
tested it, `titleLang` is a string then, so if titleLang is string do not overwrite, else overwrite